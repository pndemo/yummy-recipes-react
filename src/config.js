/*
* Application configurations
*/

import axios from 'axios';

// API URLs
export const baseAPIURL = 'http://yummy-recipes-apis.herokuapp.com';
export const authAPIURL = baseAPIURL.concat('/api/v1/auth/');
export const categoryAPIURL = baseAPIURL.concat('/api/v1/category/');
export const recipeAPIURL = baseAPIURL.concat('/api/v1/recipe/');

// axios instance for private API calls
const privateAxiosInstance = axios.create({
  headers: { Authorization: `Bearer ${localStorage.getItem('token')}` },
});
privateAxiosInstance.interceptors.request.use((config) => {
  if (localStorage.getItem('token') && config.headers.Authorization === 'Bearer null') {
    config.headers.Authorization = `Bearer ${localStorage.getItem('token')}`;
  }
  return config;
});

export default privateAxiosInstance;
