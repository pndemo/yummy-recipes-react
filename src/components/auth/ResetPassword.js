/*
* -- Reset application component --
*  Enables a registered user to request password reset
*/

import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { authAPIURL } from '../../config';
import FooterStatic from '../common/FooterStatic';
import axios from 'axios'
import { ToastContainer, toast } from 'react-toastify';

// Application styling
import '../../static/App.css';

class ResetPassword extends Component {
    // App component properties
    constructor(props) {
        super(props);
        this.state = {
            isReset: false,
            email: '',
        };
    }

    // When input changes
    onInputChanged = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    };

    // Handle password reset
    onResetClick = (event) => {
        event.preventDefault();
        axios.post(`${ authAPIURL }reset_password`,
            {
                email: this.state.email,
        })
        .then((response) => {
            if (response.status === 200) {
                this.setState({
                    isReset: true,
                })
                toast.success('Your password has been reset.')
            }
        })
        .catch((error) => {
            if (error.response.status === 400) {
                document.getElementById("error").innerHTML = error.response.data['email_message'];
            } else {
                document.getElementById("error").innerHTML = error.response.data['message'];
            }
        });
    };

    // Component rendering
    render() {
        if (this.state.isReset) {
            return <Redirect to="/login"/>
        }
        return (
            <div className="App-light">
                <div className="container-fluid">
                    <div className="row">
                        <div className="col-xs-12 col-sm-10 col-md-8 col-lg-6 col-sm-offset-1 col-md-offset-2 col-lg-offset-3">
                            <div className="App-header">
                                <h1 className="App-title-form">
                                    <span className="glyphicon glyphicon-fire"></span> Yummy Recipes
                                </h1>
                            </div>
                            <div className="panel panel-primary">
                                <div className="panel-heading text-center">
                                    Reset Password
                                </div>
                                <div className="panel-body">
                                    <form onSubmit={ this.onResetClick }>
                                        <p id="error" className="error"></p>
                                        <div className="form-group">
                                            <input
                                                className="form-control"
                                                type="text"
                                                name="email"
                                                value={ this.state.email }
                                                onChange={ this.onInputChanged }
                                                maxLength="100"
                                                placeholder="Email address"/>
                                                <p id="emailError" className="text-left error"></p>
                                        </div>
                                        <div className="form-group">
                                            <button type="submit" className="btn btn-primary btn-block">Submit</button>
                                        </div>
                                    </form>
                                </div>
                            </div>
                        </div>
                    </div>
                    <FooterStatic/>
                </div>
            </div>
        );
    }
}

export default ResetPassword;
