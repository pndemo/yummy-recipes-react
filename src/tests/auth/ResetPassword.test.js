import React from 'react';
import { mount, configure } from 'enzyme';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ResetPassword from '../../components/auth/ResetPassword';
import { authAPIURL } from '../../config';

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

// Set the mock adapter on default instance
const mock = new MockAdapter(axios);

// Mock reset password POST request
mock.onPost(`${authAPIURL}reset_password`).reply(200, { message: 'Your password has been reset.' });

// Component ResetPassword
describe('Component: ResetPassword', () => {
  configure({ adapter: new Adapter() });
  const routerComponent = mount(
  <StaticRouter location="reset_password" context={{}}>
    <ResetPassword />
  </StaticRouter>
  );

  const resetPasswordComponent = routerComponent.find('ResetPassword');
  const emailInput = resetPasswordComponent.find('[name="email"]');
  const resetPassword = resetPasswordComponent.instance();

  it('Display ResetPassword', () => {
    const rendered = renderer.create(
      <StaticRouter location="reset_password" context={{}}>
        <ResetPassword />
      </StaticRouter>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('updates state on input change', () => {
    emailInput.simulate('change', { target: { name: 'email', value: 'testuser@yummy.com' } });
    expect(resetPassword.state.email).toEqual('testuser@yummy.com');
  });

  it('reser user password', () => {
    const resetPasswordButton = resetPasswordComponent.find('[type="submit"]');
    resetPasswordButton.simulate('submit');
  });
});
