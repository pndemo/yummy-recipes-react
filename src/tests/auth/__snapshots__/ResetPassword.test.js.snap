// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResetPassword Display ResetPassword 1`] = `
<div
  className="App-light"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-10 col-md-8 col-lg-6 col-sm-offset-1 col-md-offset-2 col-lg-offset-3"
      >
        <div
          className="App-header"
        >
          <h1
            className="App-title-form"
          >
            <span
              className="glyphicon glyphicon-fire"
            />
             Yummy Recipes
          </h1>
        </div>
        <div
          className="panel panel-primary"
        >
          <div
            className="panel-heading text-center"
          >
            Reset Password
          </div>
          <div
            className="panel-body"
          >
            <form
              onSubmit={[Function]}
            >
              <p
                className="error"
                id="error"
              />
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  maxLength="100"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email address"
                  type="text"
                  value=""
                />
                <p
                  className="text-left error"
                  id="emailError"
                />
              </div>
              <div
                className="form-group"
              >
                <button
                  className="btn btn-primary btn-block"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="App-footer-static"
    >
      <p
        className="App-copyright"
      >
        Â© 2018 Yummy Recipes LLC
      </p>
    </footer>
  </div>
</div>
`;
